aws-java-sdk-core/src/main/java/com/amazonaws/ClientConfiguration.java

```diff
@@ -655,18 +699,49 @@ private String getProxyHostProperty() {
                 : getSystemProperty("http.proxyHost");
     }
 
+    /**
+     * Returns the environment variable for proxy host depending on
+     * {@link #getProtocol()}: i.e. if protocol is https, returns
+     * the host in the value of the environment variable HTTPS_PROXY/https_proxy,
+     * otherwise, returns the host in the value of the environment
+     * variable HTTP_PROXY/http_proxy.
+     */
+    private String getProxyHostEnvironment() {
+        if (getProtocol() == Protocol.HTTPS) {
+            if (getHttpsProxyUrlFromEnvironment() != null) {
+                return getHttpsProxyUrlFromEnvironment().getHost();
+            }
+        } else {
+            if (getHttpProxyUrlFromEnvironment() != null) {
+                return getHttpProxyUrlFromEnvironment().getHost();
+            }
+        }
+        return null;
```

Comment:
Can you change the logic here to match the system property logic? For example, not returning null here and catching it higher.

Reply:
Good point, I felt like this logic was a little clunky. The complication came from supporting both `HTTP_PROXY` and `http_proxy` so the logic was a little more complicated than the system property logic. The newest commit should clean it up a bit. Let me know what you think